### PR TITLE
chore(ci): add nix flake build

### PR DESCRIPTION
**Pull request by ch4og**
## Description of change
Adds flake building to CI

### **Help is needed to add rust-cache support**

### CODE REVIEW

**Comment by ch4og** [comment]
AFAIK `pkgs.rustPlatform.buildRustPackage` can't use rust-cache so I have no idea what can be done

**Comment by orhun** [comment]
Maybe @jla2000 or @Titaniumtown has an idea about this

**Comment by jla2000** [comment]
@ch4og I think you will need to cache the nix store.

**Comment by Titaniumtown** [comment]
> Maybe @jla2000 or @Titaniumtown has an idea about this

I'm not too familiar with using nix with github actions, srry :(

**Comment by ch4og** [comment]
> @ch4og I think you will need to cache the nix store.

Doesn't nix magic store do this?

**Comment by ch4og** [comment]
Should be good to merge

Caching nix store using other way worked. It uses cache after first build now

But this build time can be decreased even more with https://github.com/orhun/binsider/pull/76

File: flake.nix [hunk 1]
```diff
@@ -9,15 +9,15 @@
...
+      rec {
         packages = rec {
           binsider = pkgs.rustPlatform.buildRustPackage {
             name = "binsider";
             src = ./.;
             cargoLock = {
               lockFile = ./Cargo.lock;
             };
-            checkType = "debug";
+            buildType = "debug";
```
Comment (jla2000) [inline flake.nix:20]: Is there some specific reason why you changed the build type to debug?
I would think that most users would want a fast release build

Comment (ch4og) [reply flake.nix:20]: i agree about most users would want release but i don't see a way to build in debug mode on ci without this change in a flake

Comment (ch4og) [reply flake.nix:20]: i would be really happy to change it to release for building if someone has idea how to make it build in debug only on CI

Comment (jla2000) [reply flake.nix:20]: And why would you want to build in debug mode in CI? Just to build it faster? 
You could also just create two different packages, one that builds in debug and one in release.
And then set the default to the release one and the check to the debug one

Comment (ch4og) [reply flake.nix]: Yes, that was the reason. Thx for the tip

**Review by orhun — APPROVED** [review]
Awesome, thanks guys!

**Pull request merged**